### ISSUE TITLE
400 error 'Error: redirect_uri_mismatch'

### ISSUE THREAD

**Issue by worldsu007** [google/google-api-php-client#1406]
I tried fixing ti with post #1198 -  "Getting `redirect_uri_mismatch` error #1198" 
This did not work.  Need some help.

======================
400. That’s an error.

Error: redirect_uri_mismatch

The JavaScript origin in the request, http://bunny-6969.appspot.com, does not match the ones authorized for the OAuth client. Visit https://console.developers.google.com/apis/credentials/oauthclient/609295648701-sa2phkabvkieheivu2320jvee0ugb3od.apps.googleusercontent.com?project=609295648701 to update the authorized JavaScript origins.

Learn more

Request Details
response_type=permission id_token
scope=https://www.googleapis.com/auth/userinfo.email
openid.realm=
redirect_uri=storagerelay://http/bunny-6969.appspot.com?id=auth446392
client_id=609295648701-sa2phkabvkieheivu2320jvee0ugb3od.apps.googleusercontent.com
ss_domain=http://bunny-6969.appspot.com
gsiwebsdk=shim
========================

I have set the the redirected urls:

Authorized JavaScript origins
For use with requests from a browser. This is the origin URI of the client application. It can't contain a wildcard (https://*.example.com) or a path (https://example.com/subdir). If you're using a nonstandard port, you must include it in the origin URI.

https://bunny-6969.appspot.com 

Authorized redirect URIs
For use with requests from a web server. This is the path in your application that users are redirected to after they have authenticated with Google. The path will be appended with the authorization code for access. Must have a protocol. Cannot contain URL fragments or relative paths. Cannot be a public IP address.

http://bunny-6969.appspot.com/oauth2callback 
http://localhost:8080/oauth2callback 
https://localhost:8080/oauth2callback 
http://bunny-6969.appspot-preview.com/oauth2callback 
https://bunny-6969.appspot-preview.com/oauth2callback 
https://bunny-6969.appspot.com/oauth2callback
================================

**Comment by mattwhisenhunt** [comment]
Looks like you need to double check that what is in your php app matches what is in your Developer Console.

**Comment by LindaLawton** [comment]
The redirect URI you are sending is

>storagerelay://http/bunny-6969.appspot.com?id=auth446392
client_id=609295648701-sa2phkabvkieheivu2320jvee0ugb3od.apps.googleusercontent.com
ss_domain=http://bunny-6969.appspot.com
gsiwebsdk=shim

You cant do that.  first of all **storagerelay** will never work it needs to be a true path to a website. and you cant have any fragments on the end.    Redirect uri you are sending must exactly match the ones you have listed in Google developer console.

**Comment by mattwhisenhunt** [comment]
Thank you for filing this issue. We asked some clarifying questions or suggested a course of action a week or more ago and never heard back from you. We are unable to proceed with this issue until then, so we are closing it. Please feel free to comment with more information and we will re-open this issue.

**Comment by AlexisVidal** [comment]
need help. i have a problem, in redirect uri i typed http://localhost/hotel/AuthCallback/IndexAsync 
but when it run it looks for http://localhost/AuthCallback/IndexAsync missing hotel folder and fails cause not exist

**Comment by PatelAshish1** [comment]
In my case it was www and non-www URL. Actual site had www URL and the Authorized Redirect URIs in Google Developer Console had non-www URL. Hence, there was mismatch in redirect URI. I solved it by updating Authorized Redirect URIs in Google Developer Console to www URL.

Other common URI mismatch are:

Using http:// in Authorized Redirect URIs and https:// as actual URL, or vice-versa
Using trailing slash (http://example.com/) in Authorized Redirect URIs and not using trailing slash (http://example.com) as actual URL, or vice-versa
Here are the step-by-step screenshots of Google Developer Console so that it would be helpful for those who are getting it difficult to locate the developer console page to update redirect URIs.

1. Go to https://console.developers.google.com
2. Select your Project
3. Click on The Menu Icon
Find All step At [Here](https://icetutor.com/question/google-oauth-2-authorization-error-redirect_uri_mismatch/)

**Comment by joandylevi** [comment]
Go to https://console.developers.google.com
Select your Project
Click on The Menu Icon
Click on API Manager menu

Full Answer Available At [Here](https://icetutor.com/question/google-oauth-2-authorization-error-redirect_uri_mismatch/)

**Comment by manjeera1997** [comment]
Error 400: redirect_uri_mismatch
The redirect URI in the request, http://localhost:1410/, does not match the ones authorized for the OAuth client. To update the authorized redirect URIs, visit: https://console.developers.google.com/apis/credentials/oauthclient/554849887518-qtue6qg59evlpetkkbcmtsjjtbvnumn0.apps.googleusercontent.com?project=554849887518


I am getting this error can anyone please resolve my issue.
Thank you

**Comment by Vespiyanus** [comment]
The same issue.

**Comment by schunicd** [comment]
I am getting the error: Error 400: redirect_uri_mismatch

I understand that I need to copy and paste the uri it is displaying in the error message into the "Authorized redirect URIs" section in the Oauth2 credentials I created for my webpage.

My issue is that I am running the application locally for testing purposes, and every time I run the application it runs on a different port.

I have tried specifying the port in several different ways but no matter what I do the port changes every time.

Does anyone have any advice?
